New changes to development branch

awbarclay opened the pull request:
This is a pull request for changes I've made to the development branch since June 2.   Please test these functions for bugs and makes edits where they are needed. Also, makes suggestions if you feel something could be done differently (i.e., changes to outputs, additional/less arguments, error checks, etc.).

This is a pretty big pull request with a lot of new functions, so I'll give you over 2 weeks to complete.  Please complete your reviews by July 10th.  If you can complete your reviews sooner, that would be great!

Thanks for the help!

Andy

Commit:
CheckDupWithinSilly - added .groups = "drop_last" to summarize to aviod getting a warning message.  Also, changed bind_cols to full_join as bind_cols wasn't working as expected after dplyr updates.

Commit:
RemoveIDs - initial commit

Commit:
RemoveDups - initial commit

Commit:
AllPossibleGenotypes - Added documentation and minor changes for readablity

Commit:
CombineLoci - updated with new documentation, examples, and tidy coding.  Needs further testing. Requires dplyr 1.0

Commit:
SampSizeByLocus - updated with new documentation, examples, and tidy code.  Needs more testing.

Commit:
Plot_SampSizeByLocus - first commit for new function. Need to be tested by others for suggestions on how to make plot and maybe addtional arguments for the plot.

Commit:
create_rubias_baseline - Updated documentation and changed code for new-style "*.gcl" objects.  Baseline object needs to be tested with rubias.

Commit:
Old2new_LocCtrl - old style LocusControl objects that were updated with CombineLoci had Published locus names that were incorrect and a different length than locusnames. Added a if statement to catch this error, convert published names to NAs in the new LocusContol, and give a warning message.

Commit:
create_rubias_mixture - updated documentation, examples, and code to use new-style "*.gcl" objects. Function needs further testing and the output object and csv files need to be tested in rubias.

Commit:
old2new_gcl - added na_if("0") to replace zeros with NAs

Commit:
gcl2Genepop - complete overhaul.  Added documentation, examples, and code for new-style "*.gcl" objects.  Also added mulitcoring to speed up the function.

Commit:
FreqPop - completely rewritten to work with new tidy ".gcl" objects.  Need to test this more to see if it will break.

Commit:
FreqPop - couple of minor fixes

Commit:
Plot_Allel_Freq - all new function first commit. Still needs work and more testing.

Commit:
PoolCollections - added dplyr::all_of to selection of my.gcl to silence the warning message about using an external vector to select variables.

Commit:
CheckDupWithinSilly - Fixed bug where rubias::close_matching_samples() was throwing an error about diploid loci having NAs in the second column when loci with similar names getting sorted out of order in scores_cols. Also added subsetting for loci, which was originally intended.

Commit:
FreqPop, create_rubias_baseline, create_rubias_mixture, and gcl2Genepop - added code to ensure that the scores columns are in the correct order.

Commit:
Plot_Alle_Freq - added print() for each plot so the pdf gets populated

Commit:
FreqPop - fixed issue where the function was only calculating frequencies for dose1

Commit:
FisherCompute - updated with tidy code and created a new tibble output object. Need documentation next and further testing.

Commit:
FisherCompute - added documentation and other minor fixes.

Commit:
FisherCompute - fixed output tibble so the multiple tibbles produced in FishersTest can combined.

Commit:
FishersTest - updated with documentation and tidy coding

Commit:
FisherCompute - change code so hard zero p-values get replaced by the smallest nonnegative number that can be represented in R

Commit:
ReadGenepopPWFST - first commit of new function

Commit:
PairwiseFstTree - updated documentation and code for requiring packages

Commit:
gcl2FSTAT - new function first commit.

Commit:
PairwiseFstTree - made code more readable by adding spaces, replaced write fstat code with gcl2FSTAT(), and added package names:: where they were needed

Commit:
gcl2Genepop - added maxchar for SNPs and package names:: where they were needed

Commit:
PairwiseFstTree - added parallel connection

Commit:
PairwiseFstTree - added parallel connection

Commit:
Merge branch 'develop' of https://github.com/krshedd/GCL-R-Scripts into develop

Commit:
HoFisFstTable - added documentation, replaced fstat file code with gcl2FSTAT, added spaces, and made code tidy

Commit:
BaselineEvaSampleSizes - first commit for new function.  Needs documentation and further testing.

Commit:
gcl2FSTAT - added package names to a few functions

Commit:
CreateRubiasBaselineEval - first commit for new function. Need complete documentation and more testing.

Commit:
BaselineEvalSampleSizes - minor changes for consistency

Commit:
CreatRubiasBaselineEval - updated documentation and made minor changes for consistecy.

Commit:
CreateRubiasBaselineEval - change sapply to for loop

Commit:
run_rubias_baseline_eval - first commit for new function. Needs more testing.

Commit:
RemoveDups - added stop message when dupcheck is empty.

Commit:
custom_combine_rubias_output - added parallel loops to speed up summary when is.null(rubias_output).  Also added .groups = "drop_last" to summarize to suppress an anoying message.

Commit:
NexusColorTree - initial commit for new function

Commit:
BaselineEvalSampleSize - edited a comment

Commit:
summarize_rubias_baseline_eval - first commit for new function.  Needs further testing.

Commit:
gcl2FSTAT - added a package to pacman::p_load()

Commit:
gcl2Nexus - updated example, added some missed package names to functions, fixed some bugs, and made some minor edits for readablity

Commit:
gcl2Nexus - updated example, fixed some bugs, edited for readability, and added some missed package names to functions.

Commit:
Added stop message if special characters or spaces are detected in group_names.

Commit:
plot_baseline_eval_summary - first commit for new function. Needs further testing.

Commit:
summarize_rubias_baseline_eval - corrected some bugs and corrected one of the summary statistics

krshedd committed:
CheckDupWithinSilly - make functions explicit (package), full join by to avoid message, minor formatting

krshedd committed:
RemoveIDs - minor formating/wording changes for clarity

krshedd committed:
RemoveDups - add 'remove_both' flag to remove both duplicate IDs (ID1 and ID2) for instances where paired data has become compromised, made function calls explicit, minor formatting

krshedd committed:
AllPossiblePhenotypes - minor formatting changes

krshedd commented:
Hey @awbarclay , any reason not to have `CombinLoci.GCL` default to `update = TRUE`? Seems like we almost always want to update *LocusControl*, so that might as well be the default, right?

awbarclay commented:
Kyle,

I agree, I think that update should default to TRUE.   For some reason the default was set to FALSE in the original version.

Andy

From: Kyle Shedd <notifications@github.com>
Sent: Thursday, June 25, 2020 1:53 PM
To: krshedd/GCL-R-Scripts <GCL-R-Scripts@noreply.github.com>
Cc: Barclay, Andy W (DFG) <andy.barclay@alaska.gov>; Mention <mention@noreply.github.com>
Subject: Re: [krshedd/GCL-R-Scripts] New changes to development branch (#36)


Hey @awbarclay<https://urldefense.com/v3/__https:/github.com/awbarclay__;!!J2_8gdp6gZQ!_d7Wfo8Dw4lAoBR3DYovywyQvmNppN9feNHwJxOyrzgIKpayIY9EYn5QIQbV430agHK5ZA$> , any reason not to have CombinLoci.GCL default to update = TRUE? Seems like we almost always want to update LocusControl, so that might as well be the default, right?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/krshedd/GCL-R-Scripts/pull/36*issuecomment-649836302__;Iw!!J2_8gdp6gZQ!_d7Wfo8Dw4lAoBR3DYovywyQvmNppN9feNHwJxOyrzgIKpayIY9EYn5QIQbV432jggmoUA$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/AEJYSJDDK6XNF2KXQHUIIZ3RYPBLXANCNFSM4OGDF6AA__;!!J2_8gdp6gZQ!_d7Wfo8Dw4lAoBR3DYovywyQvmNppN9feNHwJxOyrzgIKpayIY9EYn5QIQbV433sIQPTng$>.

krshedd committed:
CombineLoci - change default to update = TRUE, clarify some language in if statements, minor formating for readability

krshedd committed:
SampSizeByLocus - default loci = LocusControl$locusnames, remove plot as input (not used), clean up some language, make functions explicit

krshedd committed:
Plot_SampSizeByLocus - default proportion limits 0 to 1, replace gather with pivot_longer, make functions explicit

krshedd committed:
old2new_LocCtrl - minor formatting changes to improve readability

krshedd committed:
old2new_gcl - make functions explicit, minor formatting changes for readability

krshedd committed:
LOKI2R - FK_FISH_ID is numeric, not character; replace gather with pivot_longer

krshedd committed:
old2new_gcl - force FK_FISH_ID as numeric and CAPTURE_DATE as_date

krshedd committed:
RemoveIndMissLoci - don't assign if no fish removed, nothing has changed

krshedd committed:
create_rubias_baseline - formatting changes

krshedd committed:
PoolCollections - minor formatting

krshedd committed:
gcl2Genepop - minor formatting, make functions explicit, standardize if/else statements

krshedd committed:
FreqPop - default loci = LocusControl$locusnames, clarify inputs, standardize if/else error catching, replace gather with pivot_longer, replace group_by + summarise with count

csjalbert commented:
Hey @awbarclay and @krshedd. Question about empty gcls ---
__SHACK07__ has 4 fish and all 4 get removed during the missing check. After removing the fish, SHACK.gcl stays in the environment, but is now empty. Should we check for remaining n and ditch any gcls that get wiped out during QC steps (missing, dups, alternates, etc.)? We would have to have some sort of warning so we know a silly was dropped from sillyvec or perhaps automatically update sillyvec... good idea, bad idea❓

Commit:
Check to see if silly is empty; if so, give more informative error msg

krshedd committed:
RemoveIDs - update check to see if silly is empty

krshedd commented:
Hey @csjalbert , ah, my bad, feel free to revert it that commit or just change it and re-commit.

I'm trying to think of a quick way to deal with this for other functions, but it seems like we'd have to add something like this to almost every function. Thoughts?

```
samp_size <- silly_n.GCL(sillyvec = sillyvec)

if(any(samp_size$n == 0)) {
  
  sillys_0 <- samp_size %>% dplyr::filter(n == 0) %>% pull(silly)
  
  stop(c("The following sillys are empty, they do not have any fish in them:\n", paste(sillys_0, collapse = "\n")))
  
}
```

csjalbert commented:
Yeah – that seems like a good and simple solution to me. I guess how useful is fixing everything? More specifically, do you think we (often-enough..) run into issues like this? I found it because the remove alternates function failed so I was troubleshooting that and realized it was the reference to the removeids within. ---edited because I didn't realize email reply pasted the entire email...

Commit:
revert changes to check for empty silly

krshedd commented:
Ya, maybe not useful enough to add to “everything”. Perhaps we could toss it at the end of RemoveMissInd and RemoveDups? The idea being that folks would run those before pretty much all other analyses?

Kyle
________________________________
From: Chase Jalbert <notifications@github.com>
Sent: Thursday, July 2, 2020 1:26:55 PM
To: krshedd/GCL-R-Scripts <GCL-R-Scripts@noreply.github.com>
Cc: Shedd, Kyle R (DFG) <kyle.shedd@alaska.gov>; Mention <mention@noreply.github.com>
Subject: Re: [krshedd/GCL-R-Scripts] New changes to development branch (#36)


Yeah – that seems like a good and simple solution to me. I guess how useful is fixing everything? More specifically, do you think we (often-enough..) run into issues like this? I found it because the remove alternates function failed so I was troubleshooting that and realized it was the reference to the removeids within.

Chase Jalbert
(907)267-2204

From: Kyle Shedd <notifications@github.com>
Sent: Thursday, July 2, 2020 1:16 PM
To: krshedd/GCL-R-Scripts <GCL-R-Scripts@noreply.github.com>
Cc: Jalbert, Chase S (DFG) <chase.jalbert@alaska.gov>; Mention <mention@noreply.github.com>
Subject: Re: [krshedd/GCL-R-Scripts] New changes to development branch (#36)


Hey @csjalbert<https://urldefense.com/v3/__https:/github.com/csjalbert__;!!J2_8gdp6gZQ!9WEqbtO0dnvDEyrpXBvNVBAZ5Vj4IwXuflvVEJpUefz-KN9iC3qEC-TlNkqWM1ZZKmy5dg$> , ah, my bad, feel free to revert it that commit or just change it and re-commit.

I'm trying to think of a quick way to deal with this for other functions, but it seems like we'd have to add something like this to almost every function. Thoughts?

samp_size <- silly_n.GCL(sillyvec = sillyvec)



if(any(samp_size$n == 0)) {



sillys_0 <- samp_size %>% dplyr::filter(n == 0) %>% pull(silly)



stop(c("The following sillys are empty, they do not have any fish in them:\n", paste(sillys_0, collapse = "\n")))



}

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/krshedd/GCL-R-Scripts/pull/36*issuecomment-653223551__;Iw!!J2_8gdp6gZQ!9WEqbtO0dnvDEyrpXBvNVBAZ5Vj4IwXuflvVEJpUefz-KN9iC3qEC-TlNkqWM1ZmnlokjA$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/AHX7BJBN74B52Y7OFWGW7K3RZT2HLANCNFSM4OGDF6AA__;!!J2_8gdp6gZQ!9WEqbtO0dnvDEyrpXBvNVBAZ5Vj4IwXuflvVEJpUefz-KN9iC3qEC-TlNkqWM1Z6_xXlcA$>.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https://github.com/krshedd/GCL-R-Scripts/pull/36*issuecomment-653227749__;Iw!!J2_8gdp6gZQ!-4o7UiWpYGhxPGVntyFbfDeuaZSF76uB31A-EyYa07DogYUYM_KYeJnaD1yEk4L0ohfd$>, or unsubscribe<https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AC54QBN3PZXIMPFDWVC4EHTRZT3R7ANCNFSM4OGDF6AA__;!!J2_8gdp6gZQ!-4o7UiWpYGhxPGVntyFbfDeuaZSF76uB31A-EyYa07DogYUYM_KYeJnaD1yEkyzJeRNq$>.

csjalbert commented:
> 
> Ya, maybe not useful enough to add to “everything”. Perhaps we could toss it at the end of RemoveMissInd and RemoveDups? The idea being that folks would run those before pretty much all other analyses? Kyle
>

Okay, that sounds good - I agree you'll likely run those first so should deal with any issues there.

Commit:
closes #38

krshedd commented:
> 
> 
> Hey @csjalbert , ah, my bad, feel free to revert it that commit or just change it and re-commit.
> 
> I'm trying to think of a quick way to deal with this for other functions, but it seems like we'd have to add something like this to almost every function. Thoughts?
> 
> ```
> samp_size <- silly_n.GCL(sillyvec = sillyvec)
> 
> if(any(samp_size$n == 0)) {
>   
>   sillys_0 <- samp_size %>% dplyr::filter(n == 0) %>% pull(silly)
>   
>   stop(c("The following sillys are empty, they do not have any fish in them:\n", paste(sillys_0, collapse = "\n")))
>   
> }
> ```

Hey guys, what do you think about creating a new function `remove_empty_collections` to check for sillys with 0 fish? Basically you'd just feed this function a *sillyvec* and it'd return a *sillyvec* with only non-zero fish sillys and print a message indicating if/which sillys had zero fish. The idea with this new function is that you could run it at any point in your analyses (ideally after QA, missing, duplicates, etc.) to scrub any non-zero fish sillys from your sillyvec. This seems like the easiest solution to me, but let me know if you have other thoughts.

awbarclay commented:
Since this issue is caused by functions that remove individuals from the analysis, and I think all of those functions call on removeIDs.GCL() to remove the fish.  Why don't we remove those collection from the workspace in that function and then have it spit out a message saying that all individuals were removed from the following sillys.  That way we don't need to add it to each function that calls on removeIDs.GCL and if it is called on by another function in the future, the issue is already taken care of.

csjalbert commented:
> 
> 
> Since this issue is caused by functions that remove individuals from the analysis, and I think all of those functions call on removeIDs.GCL() to remove the fish. Why don't we remove those collection from the workspace in that function and then have it spit out a message saying that all individuals were removed from the following sillys. That way we don't need to add it to each function that calls on removeIDs.GCL and if it is called on by another function in the future, the issue is already taken care of.

Can you think of any reason someone would NOT want them auto-removed using `removeIDs`? For example, would you want the sillyvec to stay the same throughout the QC steps then remove at the end or all okay with tossing sillys along the way? As long as it is updating sillyvec too, then I think it would be fine to toss wherever `removeIDs` is called.

awbarclay commented:
> 
> 
> > Since this issue is caused by functions that remove individuals from the analysis, and I think all of those functions call on removeIDs.GCL() to remove the fish. Why don't we remove those collection from the workspace in that function and then have it spit out a message saying that all individuals were removed from the following sillys. That way we don't need to add it to each function that calls on removeIDs.GCL and if it is called on by another function in the future, the issue is already taken care of.
> 
> Can you think of any reason someone would NOT want them auto-removed using `removeIDs`? For example, would you want the sillyvec to stay the same throughout the QC steps then remove at the end or all okay with tossing sillys along the way? As long as it is updating sillyvec too, then I think it would be fine to toss wherever `removeIDs` is called.

I can't think of any time where you would want to keep them in the analysis.  I pretty sure I've had this issue in the past with very small collections where all samples are removed, but I never had a message pop up, just the standard R error that isn't very informative.  I just remove those collections from sillyvec and continue my analysis, so I guess we don't need to remove the objects. Maybe the removeIDs function should return an object that contains the empty sillys and the user can decide whether to remove them or not.  In the QC, that object could be used to adjust the sillyvec and to account for the the empty sillys in the QC output tables. If sillyvec is automatically adjusted without the user noticing then it could cause issue later on in the analysis, e.g., groupvec will not match sillyvec.

krshedd commented:
Unfortunately not all functions use removeIDs.GCL, I can’t remember which one but either RemoveDups or RemoveIndMiss do not use removeIDs, so we’d have to add it in a couple places. I see your point about spitting out an object with the empty sillys rather than returning an object with those sillys already removed.

Kyle
________________________________
From: Andy Barclay <notifications@github.com>
Sent: Thursday, July 9, 2020 9:07:17 AM
To: krshedd/GCL-R-Scripts <GCL-R-Scripts@noreply.github.com>
Cc: Shedd, Kyle R (DFG) <kyle.shedd@alaska.gov>; Mention <mention@noreply.github.com>
Subject: Re: [krshedd/GCL-R-Scripts] New changes to development branch (#36)


Since this issue is caused by functions that remove individuals from the analysis, and I think all of those functions call on removeIDs.GCL() to remove the fish. Why don't we remove those collection from the workspace in that function and then have it spit out a message saying that all individuals were removed from the following sillys. That way we don't need to add it to each function that calls on removeIDs.GCL and if it is called on by another function in the future, the issue is already taken care of.

Can you think of any reason someone would NOT want them auto-removed using removeIDs? For example, would you want the sillyvec to stay the same throughout the QC steps then remove at the end or all okay with tossing sillys along the way? As long as it is updating sillyvec too, then I think it would be fine to toss wherever removeIDs is called.

I can't think of any time where you would want to keep them in the analysis. I pretty sure I've had this issue in the past with very small collections where all samples are removed, but I never had a message pop up, just the standard R error that isn't very informative. I just remove those collections from sillyvec and continue my analysis, so I guess we don't need to remove the objects. Maybe the removeIDs function should return an object that contains the empty sillys and the user can decide whether to remove them or not. In the QC, that object could be used to adjust the sillyvec and to account for the the empty sillys in the QC output tables. If sillyvec is automatically adjusted without the user noticing then it could cause issue later on in the analysis, e.g., groupvec will not match sillyvec.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https://github.com/krshedd/GCL-R-Scripts/pull/36*issuecomment-656245122__;Iw!!J2_8gdp6gZQ!9fLAl3jlARhWASTlIIlNOWMNPawRIyLysnpyjHrdeKOXXE874gn3KSNsWVTIH_YM2SnZ$>, or unsubscribe<https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AC54QBKCNZZLUZ6BNPXNT6TR2X2MLANCNFSM4OGDF6AA__;!!J2_8gdp6gZQ!9fLAl3jlARhWASTlIIlNOWMNPawRIyLysnpyjHrdeKOXXE874gn3KSNsWVTIH8fL9-lJ$>.

awbarclay commented:
Yes, we should use removeIDs in all functions that remove fish from *.gcl's,  I probably wrote that function after RemoveIndMissLoci.  I'm pretty sure that RemoveDups calls on it, but we should check to confirm.

csjalbert commented:
> 
> 
> Yes, we should use removeIDs in all functions that remove fish from *.gcl's, I probably wrote that function after RemoveIndMissLoci. I'm pretty sure that RemoveDups calls on it, but we should check to confirm.

just checked - `removedups` does call on `removeids`. `RemoveIndMissLoci `does not

krshedd commented:
Sounds good, so we should change RemoveIDs to flag when a silly gets emptied and we need to change RemoveIndMiss to use RemoveIDs, right? Who wants to take what?

From: Chase Jalbert <notifications@github.com>
Sent: Thursday, July 9, 2020 9:42 AM
To: krshedd/GCL-R-Scripts <GCL-R-Scripts@noreply.github.com>
Cc: Shedd, Kyle R (DFG) <kyle.shedd@alaska.gov>; Mention <mention@noreply.github.com>
Subject: Re: [krshedd/GCL-R-Scripts] New changes to development branch (#36)


Yes, we should use removeIDs in all functions that remove fish from *.gcl's, I probably wrote that function after RemoveIndMissLoci. I'm pretty sure that RemoveDups calls on it, but we should check to confirm.

just checked - removedups does call on removeids. 'RemoveIndMissLoci` does not

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/krshedd/GCL-R-Scripts/pull/36*issuecomment-656261284__;Iw!!J2_8gdp6gZQ!5KIWELikXpJcIVH8akw3QWiXPogSyvIALgGSxiBYIz6pbL5Ap9fmmvNJftDpu1JQ6zwy$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/AC54QBLRJ6KFGC6W4UEPZ2LR2X6PFANCNFSM4OGDF6AA__;!!J2_8gdp6gZQ!5KIWELikXpJcIVH8akw3QWiXPogSyvIALgGSxiBYIz6pbL5Ap9fmmvNJftDpu0Syav-e$>.

awbarclay commented:
That sounds right Kyle.  Unless Chase wants to do it, I can make the changes to removeIDs and add it to RemoveIndMissLoci.

krshedd commented:
Awesome, if you have time, that would be great. I’ll keep plugging away at the other functions.

From: Andy Barclay <notifications@github.com>
Sent: Thursday, July 9, 2020 10:50 AM
To: krshedd/GCL-R-Scripts <GCL-R-Scripts@noreply.github.com>
Cc: Shedd, Kyle R (DFG) <kyle.shedd@alaska.gov>; Mention <mention@noreply.github.com>
Subject: Re: [krshedd/GCL-R-Scripts] New changes to development branch (#36)


That sounds right Kyle. Unless Chase wants to do it, I can make the changes to removeIDs and add it to RemoveIndMissLoci.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/krshedd/GCL-R-Scripts/pull/36*issuecomment-656292446__;Iw!!J2_8gdp6gZQ!9rzO0G5J520YCXw31oAOHrWTjZqsiXuQNfFays0uDuJ9snh-E-UsANnjNXlBlcV4Uoz6$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/AC54QBLGIN3LOXL6BPNQMMTR2YGNPANCNFSM4OGDF6AA__;!!J2_8gdp6gZQ!9rzO0G5J520YCXw31oAOHrWTjZqsiXuQNfFays0uDuJ9snh-E-UsANnjNXlBlXqnG6Qq$>.

awbarclay commented:
Okay, I'm on it.  It shouldn't take too long.

krshedd commented:
Awesome, thanks.

From: Andy Barclay <notifications@github.com>
Sent: Thursday, July 9, 2020 10:54 AM
To: krshedd/GCL-R-Scripts <GCL-R-Scripts@noreply.github.com>
Cc: Shedd, Kyle R (DFG) <kyle.shedd@alaska.gov>; Mention <mention@noreply.github.com>
Subject: Re: [krshedd/GCL-R-Scripts] New changes to development branch (#36)


Okay, I'm on it. It shouldn't take too long.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/krshedd/GCL-R-Scripts/pull/36*issuecomment-656294131__;Iw!!J2_8gdp6gZQ!49xCqBzwYKEXhwg0hRzImiejFP1C4UAroLG5UlpclMIqKOARwhqgRzFGtReLdSHtETQR$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/AC54QBJFKGGU3WBRN2E3RPDR2YG3XANCNFSM4OGDF6AA__;!!J2_8gdp6gZQ!49xCqBzwYKEXhwg0hRzImiejFP1C4UAroLG5UlpclMIqKOARwhqgRzFGtReLdbVUoZ_H$>.

csjalbert commented:
Awesome - thanks, Andy. I am just picking up the baseline conversion to new / function review so will update anything I come across with that.

Commit:
removed references to ncores from header since it isn't actually an argument or used.

Commit:
Added new output tibble to RemoveIDs.GCL that includes a column indicating if all IDs were removed (is_empty) and a warning message about removing all IDs from the silly.

Commit:
Updated RemoveMissLoci.GCL to output new RemoveIDs output tibble.

Commit:
Updated RemoveDups.GCL to include new RemoveIDs output tibble.

Commit:
fixed example and intro; spacing for readability; added x and y text size; added popnames var to allow for renaming pops;

Commit:
Merge branch 'develop' of https://github.com/krshedd/GCL-R-Scripts into develop

Commit:
Added example dir and QCExample data to use during QC steps

Commit:
rework to gcl 2.0 standards; added example data

Commit:
RemoveDups - changed stop message to a warning if dupcheck is an empty data frame and added a line to return an empty data frame.

Commit:
run_rubias_mixture - added pload for required packages and stop message if reference and mixture objects have different header names. Removed "hoser!!!" from group_names stop message.

Commit:
AttributesToIDs is no longer needed with the new-style .gcl objects.

Commit:
PoolCollection - fixed example

Commit:
Plot_Allele_Freq - added a new argument to adjust x-axis tick interval when popnames=NULL. Made minor changes for readablity and suppressed ggplot warnings caused by NAs.

Commit:
added missing + after xlab

Commit:
Fixed example

Commit:
minor correction to documentation

Commit:
Made documentation more consistent between gcl2FSTAT and gcl2GENEPOP

Commit:
Minor edits to documentation

Commit:
Minor edit to documentation

Commit:
initial commit - Kyle's fancy function to view pooling test results.

Commit:
changed to facet_wrap; added section to auto-calculate ncols

Commit:
Minor edits for consistency, edits to input documentation, and added empty output and warning when no idividuals are found to be the wrong species.

Commit:
Merge branch 'develop' of https://github.com/krshedd/GCL-R-Scripts into develop

Commit:
initial commit - plotting HWE from ReadGenepopHWE

Commit:
created loci var and removed hard-coded loci

Commit:
explicit function calls

Commit:
added a package call

csjalbert commented on plot_FishersTest.GCL.R:
```diff
@@ -0,0 +1,46 @@
+    ) +
+    ggplot2::facet_wrap(~ test_sillys, ncol = ncols) +
+    ggplot2::theme_bw()
+}
```
I added this and it looks like it's wrapped into the PR @awbarclay - may want to test this one out

csjalbert commented on plot_GenepopHWE.GCL.R:
```diff
@@ -0,0 +1,49 @@
+    ) +
+    ggplot2::facet_wrap(~ silly, ncol = ncols) +
+    ggplot2::theme_bw()
+}
```
I added this and it looks like it's wrapped into the PR @awbarclay and @krshedd - This is a rough draft and likely needs edits, I didn't realize it would be pushed into the open PR.

Commit:
removed loci argument - automatically calculated from the pooling_test object now

Commit:
rm loci argument - calculates it from GenepopHWE_report, otherwise 'loci' and genepop reported loci may differ

Commit:
rm loci argument - calculates it from GenepopHWE_report, otherwise 'loci' and genepop reported loci may differ

Commit:
Merge branch 'develop' of https://github.com/krshedd/GCL-R-Scripts into develop

# Conflicts:
#	plot_GenepopHWE.GCL.R

Commit:
PoolCollections - added code to make sure that the SILLY_CODE column in the new pooled silly contains the new silly name (newname)

Commit:
CheckDupsWithinSilly - updated examples to use a pooled collection and changed code when quantile = NULL so that it unites SILLY_CODE and FK_FISH_ID to create the indiv variable. This update was needed after changing PoolCollections.GCL to make SILLY_CODE = newname

Commit:
CheckDupWithinSilly - added regular expression to separate indiv by last instance of an underscore.

Commit:
CheckDupWithinSilly and RemoveAlternates- fixed regular expression to separate indiv by last instance of an underscore.

Commit:
RemoveAlternateSpecies - added regular expression to separate at last underscore on line 31

Commit:
fixed regex from previous commit

Merge branch 'develop' of https://github.com/krshedd/GCL-R-Scripts into develop

# Conflicts:
#	CheckDupWithinSilly.GCL.r
#	RemoveAlternateSpecies.GCL.r

Commit:
fixed regex from previous commit(s); edited example

Commit:
summarize_rubias_baseline_eval - updated path for example and corrected the regular expression to separate test_group and scenario.

Commit:
added \n so warning messages appear on sep. lines

Commit:
Added Within_Interval to output of summarize_rubais_baseline_eval and to plots produced by plot_baseline_eval_summary

Commit:
Minor change to summarize_rubias_baseline_eval

Commit:
plot_baseline_eval_summary - factored test_groups before plotting to ensure that the order of groups is preserved.

Commit:
init. commit of the tissue location mapper function

Commit:
expanded example to include exporting of maps, all tissues, incorrect tissues

Commit:
fixed tissue_data and tissuemap calculations; clarified example; spelling fixes

Commit:
keep exhausted tissues

Commit:
fixed inputs getting overwritten by vars; added write_csv() to V: location; added anti_join() to tissue mapping so duplicate locations are not copied onto maps

Commit:
correct function name and source location

Commit:
reformatted data to remove dashes

krshedd reviewed (approved):
Looks good to me!
